### PR TITLE
make `BagsList::put_in_front_of` be permissionless

### PR DESCRIPTION
**Pull request by kianenigma**
This creates a permissionless variant of the `put_in_front_of`, allowing anyone to move users within a bag. 

Learn more about bags and put-in-front-of: https://polkadot.network/blog/staking-update-january-2022#put-in-front-of-extrinsic

No operation is free in this process. 

Moving within the bag is only possible if you are being strictly moved to a better position. But there is no guarantee that you will be moved to _the best_ possibly position.

I kept the old transaction untouched to not break any existing UIs. 

This new transaction should be integrated into https://github.com/paritytech/polkadot-scripts, and run as a bot by w3f.

### CODE REVIEW

**Review by gpestana — APPROVED** [review]
I'd probably just add a simple test for the new extrinsic just to ensure that future changes do not affect the wanted behaviour.

**Review by muharem — APPROVED** [review]

File: frame/bags-list/src/lib.rs [hunk 1]
```diff
@@ -257,6 +259,24 @@ pub mod pallet {
 				.map_err::<Error<T, I>, _>(Into::into)
 				.map_err::<DispatchError, _>(Into::into)
 		}
+
+		/// Same as [`put_in_front_of`], but it can be called by anyone.
+		///
+		/// Fee is payed by the origin under all circumstances.
```
Comment (liamaharon) [inline frame/bags-list/src/lib.rs:265]:
```suggestion
		/// Fee is paid by the origin under all circumstances.
```

**Review by liamaharon — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3370230

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3370229

**Comment by kianenigma** [comment]
@snowmead @wentelteefje @aaronbassett can you do a quick review in terms of doc? This is the type of work I would love to see more of: Make a small contribution to Pallet, and use it as an excuse to learn and improve its documentation. Ofc this won't scale to all pallets in the universe, but it should help us improve the docs of the more heavily used, and recently developed ones.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate-macos
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3370327

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3370379

File: frame/bags-list/Cargo.toml [hunk 2]
```diff
@@ -27,6 +27,8 @@ frame-election-provider-support = { version = "4.0.0-dev", default-features = fa
 
 # third party
 log = { version = "0.4.17", default-features = false }
+docify = "0.2.0"
```
Comment (sam0x17) [inline frame/bags-list/Cargo.toml:30]:
```suggestion
docify = "0.2.1"
```
updated a few days ago to fix oliver's issue

**Review by sam0x17 — APPROVED** [review]

**Review by snowmead — APPROVED** [review]

**Comment by kianenigma** [comment]
bot merge

**Pull request merged**